bin/ch/ch.cpp

```diff
@@ -671,6 +671,10 @@ HRESULT ExecuteTest(const char* fileName)
         }
     }
 Error:
+    if( fileContents != nullptr)
+    {
+        free((char*)fileContents);
```

Comment:
Yeah, that's what I'm doing in the attempt that I added, and I shoehorned in some other data, but it only really works because I know externally to the actual code that we're completely done with the source buffer and all data needed to create it. Let's say we have a zip file containing both cached bytecode and source code as separate sections, and we want to create a function to pass around with JsParseSerializedScript. While we can use this function to have a loadCallback, and in that return an ExternalArrayBuffer with a finalizeCallback, we don't have a good way to know "we no longer need the ability to construct the source buffer for this, so you can free the data needed to construct it", which would let us free the memory being taken up by the compressed version of the source code. Here we only get away with it because we're using JsRunSerializedScript, and I know afterwards that we're not going to do any more calls so I can free the buffer that would be used to construct the ExternalArrayBuffer if it were to be needed, but in more complex cases I think we'd need another callback or something to properly handle the scoping.